### PR TITLE
ETH-717: added minimum undelegation seconds to StreamrConfig

### PR DESCRIPTION
**Pull request by jtakalai**

### CODE REVIEW

File: packages/network-contracts/contracts/OperatorTokenomics/OperatorPolicies/DefaultUndelegationPolicy.sol [hunk 1]
```diff
@@ -27,17 +27,23 @@ contract DefaultUndelegationPolicy is IUndelegationPolicy, Operator {
      * @dev   Minimum delegation is handled when paying out the queue, and checked separately on _transfer.
      **/
     function onUndelegate(address delegator, uint amountDataWei) external {
-        // limitation only applies to the operator, others can always undelegate
```
Comment (hpihkala) [inline packages/network-contracts/contracts/OperatorTokenomics/OperatorPolicies/DefaultUndelegationPolicy.sol:30]: Could this change be more minimal?

File: packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts [hunk 2]
```diff
@@ -701,6 +702,30 @@ describe("Operator contract", (): void => {
             await expect(operator.undelegate(parseEther("1000000")))
                 .to.emit(operator, "Undelegated").withArgs(operatorWallet.address, parseEther("10000"))
         })
+
+        it("prevents too fast undelegation", async function(): Promise<void> {
+            // TODO: fix tests to tolerate undelegation limit, then remove this
```
Comment (hpihkala) [inline packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts:707]: How about this todo?

**Comment by hpihkala** [comment]
Additional question: can OperatorTokens be transferred to work around the checks?

File: packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts (hunk 2 above)
Comment (jtakalai) [reply packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts:707]: later

Comment (jtakalai) [reply packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts:707]: I'll make a ticket

**Comment by jtakalai** [comment]
> Additional question: can OperatorTokens be transferred to work around the checks?

seems they can not, after all! UndelegationPolicy is checked for the sender already.

File: packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts [hunk 3]
```diff
@@ -701,6 +702,64 @@ describe("Operator contract", (): void => {
...
+            await expect(operator.undelegate(parseEther("500"))).to.be.revertedWith("error_undelegateTooSoon")
+            await expect(operator.connect(delegator).undelegate(parseEther("500"))).to.be.revertedWith("error_undelegateTooSoon")
+
+            await advanceToTimestamp(timeAtStart + 3000, "Try again to undelegate")
+            await expect(operator.connect(delegator).undelegate(parseEther("500")))
+                .to.emit(operator, "Undelegated").withArgs(delegator.address, parseEther("500"))
+            await expect(operator.undelegate(parseEther("500")))
+                .to.emit(operator, "Undelegated").withArgs(operatorWallet.address, parseEther("500"))
+
+            // TODO: fix tests to tolerate undelegation limit, then remove this
```
Comment (jtakalai) [inline packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts:726]: there is a ticket for this

**Review by samigabor — APPROVED** [review]
LGTM 
One note though: the old undelegation policy must be kept as trusted otherwise for the existing operators undelegation would be broken

**Review by samt1803 — APPROVED** [review]
I didn’t execute it but from looking it over everything looks good and it should solve the problem nicely 👍

**Pull request merged**